src/platform/i386/chal_pgtbl.h

```diff
@@ -0,0 +1,192 @@
+#ifndef CHAL_PGTBL_H
+#define CHAL_PGTBL_H
+
+/* These code below are for x86 specifically, only used in x86 chal */
+typedef enum {
+	X86_PRESENT    = 1,
+	X86_WRITABLE   = 1 << 1,
+	X86_USER       = 1 << 2,
+	X86_WT         = 1 << 3, /* write-through caching */
+	X86_NOCACHE    = 1 << 4, /* caching disabled */
+	X86_ACCESSED   = 1 << 5,
+	X86_MODIFIED   = 1 << 6,
+	X86_SUPER      = 1 << 7, /* super-page (4MB on x86-32) */
+	X86_GLOBAL     = 1 << 8,
+	/* Composite defined bits next*/
+	X86_COSFRAME   = 1 << 9,
+	X86_COSKMEM    = 1 << 10, /* page activated as kernel object */
+	X86_QUIESCENCE = 1 << 11,
+	/* Flag bits done. */
+
+	X86_USER_DEF   = X86_PRESENT | X86_USER | X86_ACCESSED | X86_MODIFIED | X86_WRITABLE,
+	X86_INTERN_DEF = X86_USER_DEF,
+} pgtbl_flags_x86_t;
+
+/**
+ * Use the passed in page, but make sure that we only use the passed
+ * in page once.
+ */
+static inline void *
+__pgtbl_a(void *d, int sz, int leaf)
```

Comment:
should we namespace to `chal_pgtbl`?

Reply:
I don't think so, as these are not public APIs.